lib/libc/stdlib/malloc/snmalloc/Makefile.inc

```diff
@@ -0,0 +1,48 @@
+.PATH: ${LIBC_SRCTOP}/stdlib/malloc/snmalloc
+
+MISRCS+=	malloc.cc
+
+# Specify the locations in contrib for headers.
+CXXFLAGS+=	-I${SRCTOP}/contrib/subrepo-snmalloc/src/snmalloc
+
+CXXSTD?=	c++20
+
+# C++ dialect options.
+CXXFLAGS+=	-fno-exceptions -fno-rtti
+.if ${MACHINE_ARCH} == amd64
+# -mcx16 is needed to avoid calls to libatomic.
+CXXFLAGS+=	-mcx16
```

Comment:
Seems fraught upstream, but I'm not convinced we care for CheriBSD (even that v2 Atom microarchitectures are a decade old at this point) and the code doesn't build otherwise.

Reply:
For upstream FreeBSD it would be a negotiation, though it might be more sellable if snmalloc is not the default.  CX16 is present on a pre-Nehalem system I have access to.  It's older than SSE4.x so it's not that hard of a sell to require, though requiring it for libc is a pretty big ask.

Reply:
If it's a performance issue then snmalloc will suck on things like RISC-V unless 16-byte atomics are avoided. If it's not a performance issue then the underlinking should be fixed.

Reply:
I wondered about this, but it seems snmalloc assumes it can use plain LR/SC to avoid ABA and only uses double-CAS on arches without LR/SC where an extra generation count is used to prevent ABA.

Reply:
It's not underlinking that's blocking me (the comment is at least one step removed from the actual issue). The upstream snmalloc x86 platform code refuses to compile without `__GCC_HAVE_SYNC_COMPARE_AND_SWAP_16`. https://github.com/microsoft/snmalloc/blob/7b597335aedc2c60b0174f6e67a7f7560f841760/src/snmalloc/ds/aba.h#L108. I've updated the comment and will file an issue.